components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/keystore/KeyStoreManagementServiceImpl.java

```diff
@@ -128,8 +130,8 @@ public X509Certificate getClientCertificate(String tenantDomain, String alias) t
 
         KeyStore trustStore = null;
         try {
-            trustStore = getKeyStoreAdmin(tenantDomain).getTrustStore();
-        } catch (SecurityConfigException e) {
+            trustStore = getKeyStoreManager(tenantDomain).getTrustStore();
```

Comment:
Shall we not change the KeyStoreAdmin class usages since we are planning to introduce a new constructor by removing the registry initializations.